docs/DeveloperGuide.adoc

```diff
@@ -447,6 +447,73 @@ In other words, when RestaurantBook#deleteMember() or RestaurantBook#deleteIngre
 
+=== The `stats` command
+
+When the `stats` command is entered, the current statictis of the amount of bookings is to be displayed to the user in the form of a bar chart.
+
+==== Current implementation
+
+The `RestaurantBookParser` class would detect if the user has keyed in the keyword `stats`. If it does, `RestaurantBookParser` would call the `Statistics` class. The `Statistics` class is used to determine what data is to be displayed to the user in the bar chart (ie what are the axes, values, and legends of the bar chart). It would call the `StatsWindow` class which handles the UI. The bar chart will appear in a new window and does not affect the current window.
+
+==== Design Considerations
+
+===== Aspect: Implementation of data structure
+
+* **Alternative 1 (current choice):** Doing the calculation only when the user wants to view the statistics.
+** Pros: Calculating only when needed is more efficient.
+** Cons: The command would take some time to calculate before being able to display the statistics.
+
+* **Alternative 2:** Every change to the database updates the statistics on the fly.
+** Pros: Displaying the statistics on the UI would be faster.
+** Cons: Need to change the implementation of every command that causes a change to the database. Calculations have to be done every operation, affecting efficiency.
+
+* Decision: Alternative 1 was adopted as it is an easier implementation. As the number of times the `stats` command is called is likely to be low as compared to an `add` or `delete` operation, doing the calculation after every operation would be a waste of computation power. 
+
+===== Aspect: Design of UI display
+
+* **Alternative 1 (current choice):** Displaying the statistics in a new window.
+** Pros: The statistics would be displayed independently from the database and would not interfere with each other. 
+** Cons: The user can open multiple windows which contains different data and not know which is the correct window.
+
+* **Alternative 2:** Hide the displayed database to show the statistics.
+** Pros: There would be only 1 window to manage at any point in time.
+** Cons: Cannot view the statistics and the database concurrently.
+
+* Decision: Alternative 1 was adopted as it is an easier implementation. Implementing the `stats` window as a static window that will not change would not affect the other parts of the app, and there is no need to hide and show the current screen. Moreover, the user might want to view the statistics together with the database, which is only possible with multiple windows.
```

Comment:
Easier to implement- not mentioned in the pros and cons analysis? Try not to suddenly bring in a new point like that- bad logical flow.